main/file.c

```diff
@@ -147,26 +167,65 @@ sub_status_t sub_init(sub_init_struct_t *init_struct, uint16_t transmit_periods[
     return SUB_OK;
 }
 
+static void set_transmit_state(bool state){
+
+    if(xSemaphoreTake(sub_struct.transmit_time_mutex, portMAX_DELAY) == pdTRUE){
+
+        sub_struct.transmit_state = state;
+
+        xSemaphoreGive(sub_struct.transmit_time_mutex);
+    }
+
+}
+
+static void transmit_timer_cb(TimerHandle_t xTimer){
+
+    //TODO: POZBYĆ SIĘ SEMAFORA Z CALLBACKA 
+    //W LORZE JEST TASK NOTIFY I PEWNIE TO JEST GIT POMYSŁ
+    set_transmit_state(true);
+}
+
 
 sub_status_t sub_get_rx_data(uint8_t *data_buffer, size_t buffer_size){
-    return 0;
+    
```

Comment:
Tak naprawdę ta funkcja pozwala na odczytanie tylko i wyłącznie komendy, bo tylko one są pchane na kolejkę. Nie ma sensu więc kopiować do bufforu, z którego trzeba będzie to później odkodowywać, tylko od razu tutaj jako argument można przyjąć wskaźnik na recv_cb_cmd_t i do niego wrzucać